object-curly-newline counts default export in minProperties

**Tell us about your environment**

Ubuntu 17.10

* **ESLint Version:** 4.19.0
* **Node Version:** 9.6.1
* **npm Version:** 5.7.1

(Also validated on OSX, Node 7.10.0, npm 4.2.0)

**What parser (default, Babel-ESLint, etc.) are you using?**

Default, I believe(?), with sourceType: module. Full configuration:

<details>
<summary>Configuration</summary>

`.eslintrc`
```json
{
	"root": true,
	"env": { "es6": true },
	"extends": [
		"eslint:recommended"
	],
	"parserOptions": {
		"sourceType": "module"
	},
	"rules": {
		"object-curly-newline": [ "error", {
			"minProperties": 2,
			"consistent": true
		} ]
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[This gist](https://gist.github.com/kadamwhite/f0098c63df2f9edf5d79b8b8b8b446f5) contains the `package.json`, `.eslintrc` and `test.js` as described here; downloading that gist, running `npm install` and then `npm test` replicates the issue on every environment I can access.

<!-- Paste the source code below: -->
`test.js`
```js
import DefaultExport, { NamedExport } from 'some-module';
import { NamedExportWithoutDefaultImport } from 'some-other-module';

export default {
  DefaultExport,
  NamedExport,
  NamedExportWithoutDefaultImport,
};
```
`package.json`
```json
{
  "name": "test-object-curly-newline",
  "scripts": {
    "test": "eslint ./test.js"
  },
  "devDependencies": {
    "eslint": "^4.19.0"
  }
}
```
<!-- Paste the command you used to run ESLint: -->
```bash
npm test # runs `eslint ./test.js`
```

**What did you expect to happen?**

I expected no `object-curly-newline` issue to be reported for either of these imports, as they both have only a single named export within curly brackets:
```js
import DefaultExport, { NamedExport } from 'some-module';
import { NamedExportWithoutDefaultImport } from 'some-other-module';
```
however the default export on the first line is being counted as an object property, meaning that the rule matches the `minProperties` and I get errors for each curly bracket on Line 1, but not Line 2

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ npm test

> test-object-curly-newline@ test /hm/gh/coding-standards/packages/eslint-config-humanmade/test
> eslint ./test.js


/hm/gh/coding-standards/packages/eslint-config-humanmade/test/test.js
  1:23  error  Expected a line break after this opening brace   object-curly-newline
  1:37  error  Expected a line break before this closing brace  object-curly-newline

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

npm ERR! Test failed.  See above for more details.
```

--- a/lib/rules/object-curly-newline.js
+++ b/lib/rules/object-curly-newline.js
@@ -116,7 +116,8 @@ function areLineBreaksRequired(node, options, first, last) {
     } else {
 
         // is ImportDeclaration or ExportNamedDeclaration
-        objectProperties = node.specifiers;
+        objectProperties = node.specifiers
+            .filter(s => s.type === "ImportSpecifier" || s.type === "ExportSpecifier");
     }
 
     return objectProperties.length >= options.minProperties ||

--- a/tests/lib/rules/object-curly-newline.js
+++ b/tests/lib/rules/object-curly-newline.js
@@ -493,6 +493,10 @@ ruleTester.run("object-curly-newline", rule, {
             ].join("\n"),
             options: [{ ImportDeclaration: { minProperties: 3 } }]
         },
+        {
+            code: "import DefaultExport, {a} from 'module';",
+            options: [{ ImportDeclaration: { minProperties: 2 } }]
+        },
 
         // "ExportDeclaration" ---------------------------------------------
         {
@@ -1588,6 +1592,19 @@ ruleTester.run("object-curly-newline", rule, {
             errors: [
                 { line: 1, column: 8, message: "Unexpected line break after this opening brace." },
                 { line: 3, column: 1, message: "Unexpected line break before this closing brace." }
+            ]
+        },
+        {
+            code: "import DefaultExport, {a, b} from 'module';",
+            output: [
+                "import DefaultExport, {",
+                "a, b",
+                "} from 'module';"
+            ].join("\n"),
+            options: [{ ImportDeclaration: { minProperties: 2 } }],
+            errors: [
+                { line: 1, column: 23, message: "Expected a line break after this opening brace." },
+                { line: 1, column: 28, message: "Expected a line break before this closing brace." }
             ]
         },
 

